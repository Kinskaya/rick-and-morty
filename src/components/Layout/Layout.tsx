import React from 'react';
import { NavLink, Outlet } from 'react-router-dom';

class Layout extends React.Component {
  render() {
    return (
      <>
        <header>
          <NavLink to="/">Home</NavLink>
          <NavLink to="/about">About Us</NavLink>
          <NavLink to="/order">Order</NavLink>
          <NavLink to="/:id" onClick={(event) => event.preventDefault()}>
            Card
          </NavLink>
        </header>
        <main className="container">
          <Outlet />
        </main>
      </>
    );
  }
}
export default Layout;
