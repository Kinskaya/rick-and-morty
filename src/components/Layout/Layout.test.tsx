import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Layout from '../Layout/Layout';

describe('Layout component', () => {
  it('render correctly', () => {
    const { asFragment } = render(<Layout />, { wrapper: BrowserRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
