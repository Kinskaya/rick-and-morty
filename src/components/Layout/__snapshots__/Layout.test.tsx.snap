// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component render correctly 1`] = `
<DocumentFragment>
  <header>
    <a
      aria-current="page"
      class="active"
      href="/"
    >
      Home
    </a>
    <a
      class=""
      href="/about"
    >
      About Us
    </a>
    <a
      class=""
      href="/order"
    >
      Order
    </a>
  </header>
  <main
    class="container"
  />
</DocumentFragment>
`;
